exporter/collector/monitoredresource.go

```diff
@@ -123,37 +130,37 @@ func (m *metricMapper) resourceMetricsToMonitoredResource(
 
 	switch cloudPlatform {
 	case semconv.AttributeCloudPlatformGCPComputeEngine:
-		return createMonitoredResource(gceInstance, attrs)
+		return m.createMonitoredResource(gceInstance, attrs)
 	case semconv.AttributeCloudPlatformGCPKubernetesEngine:
 		// Try for most to least specific k8s_container, k8s_pod, etc
 		if _, ok := attrs.Get(semconv.AttributeK8SContainerName); ok {
-			return createMonitoredResource(k8sContainer, attrs)
+			return m.createMonitoredResource(k8sContainer, attrs)
 		} else if _, ok := attrs.Get(semconv.AttributeK8SPodName); ok {
-			return createMonitoredResource(k8sPod, attrs)
+			return m.createMonitoredResource(k8sPod, attrs)
 		} else if _, ok := attrs.Get(semconv.AttributeK8SNodeName); ok {
-			return createMonitoredResource(k8sNode, attrs)
+			return m.createMonitoredResource(k8sNode, attrs)
 		}
-		return createMonitoredResource(k8sCluster, attrs)
+		return m.createMonitoredResource(k8sCluster, attrs)
 	case semconv.AttributeCloudPlatformAWSEC2:
-		return createMonitoredResource(awsEc2Instance, attrs)
+		return m.createMonitoredResource(awsEc2Instance, attrs)
 	default:
 		// Fallback to generic_task
 		_, hasServiceName := attrs.Get(semconv.AttributeServiceName)
 		_, hasServiceInstanceID := attrs.Get(semconv.AttributeServiceInstanceID)
 		if hasServiceName && hasServiceInstanceID {
-			return createMonitoredResource(genericTask, attrs)
+			return m.createMonitoredResource(genericTask, attrs)
 		}
 
 		// If not possible, fallback to generic_node
-		return createMonitoredResource(genericNode, attrs)
+		return m.createMonitoredResource(genericNode, attrs)
```

Comment:
Config changes LGTM. It will allow config to override the source labels, but there is still no way for them to override the monitored resource type selection in the first place (hence the changes to GKE control plane fixture didn't work).

Any plans for how to do this?

Reply:
I could be wrong about the fixture. But I think there still needs to be a way to override the MR type selection

Reply:
I fixed it.  We can discuss design below.